### PR TITLE
Icon react component

### PR DESCRIPTION
**Pull request by matikrk**
* Create icon component with corresponding tests
* Upgrade DockBlock.jsx
* Replace old html page to new one created in jsx
* add fb icon to types
* use Icon component in Button examples

### CODE REVIEW

File: src/components/icons/Icon.jsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+const types = {
```
Comment (aju) [inline src/components/icons/Icon.jsx]: simplify it just types = {'answer', 'answered' ... }

File: src/components/icons/Icon.jsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  blue: 'blue',
+  mustard: 'mustard',
+  lavender: 'lavender',
+  peach: 'peach',
+  dark: 'dark',
+  mint: 'mint'
+};
+const sizes = [120, 118, 96, 94, 64, 62, 48, 46, 38, 32, 30, 26, 24, 22, 20, 18, 16, 14, 10, 8];
+
+const Icon = ({color, size, type}) => {
```
Comment (aju) [inline src/components/icons/Icon.jsx]: some default size maybe ?

File: src/components/icons/Icon.jsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  report_flag: 'report_flag',
+  search: 'search',
+  star: 'star',
+  stream: 'stream',
+  symbols: 'symbols',
+  unseen: 'unseen',
+  x: 'x',
+  fb: 'fb'
+};
+const colors = {
```
Comment (vergilius) [inline src/components/icons/Icon.jsx]: shouldn't keys be uppercase'd? wdyt?

File: src/components/icons/icon.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  const icon = shallow(
+    <Icon type={types.answer}/>
+  );
+
+  expect(icon.hasClass('sg-icon')).toEqual(true);
+  expect(icon.find('use')).toHaveLength(1);
+});
+
+test('error when no type', () => {
+  const spy = jest.spyOn(console, 'error');
```
Comment (vergilius) [inline src/components/icons/icon.spec.jsx]: I don't know jest, is this global that is injected?

File: src/components/icons/pages/icons.jsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import DocsBlock, {contrastBlockCssClass} from '../../../docs/DocsBlock';
```
Comment (vergilius) [inline src/components/icons/pages/icons.jsx]: `../../../` its easy to get lost, but I suppose there are more pros than cons

File: src/components/icons/Icon.jsx (hunk 1 above)
Comment (kdzwinel) [inline src/components/icons/Icon.jsx]: +1

File: src/components/icons/icon.spec.jsx (hunk 4 above)
Comment (matikrk) [inline src/components/icons/icon.spec.jsx]: jest contains few libraries for testing and you don't need define everything. Variables test, expect, describe is defined by jest globally. For testing React we need only import enzyme library with one allow fast rendering components for tests

File: src/components/icons/pages/icons.jsx (hunk 5 above)
Comment (matikrk) [inline src/components/icons/pages/icons.jsx]: DocsBlock it's internal component for build docs, we don't export it like other components

**Pull request merged**